### PR TITLE
Update build for 2024.2 support

### PR DESCRIPTION
**Pull request by AkibaAT**
## The Problem/Issue/Bug:
Resolves issue #347 

## How this PR Solves the Problem:
Raise platformVersion to 2024.2
Upgrade intellij-platform-gradle-plugin to 2.0.1
Upgrade Java to 21

## Manual Testing Instructions:
./gradlew build
Install resulting build/distributions/ddev-intellij-plugin-0.0.1-dev.zip
Test functionality in PhpStorm / IntelliJ

### CODE REVIEW

**Comment by thomaskieslich** [comment]
Hi, it works under macos-latest and 2024.2. 
Error java.io.IOException: Connection failed: too many redirects still exists.
Will try to combine it with the other pull request.

**Comment by AkibaAT** [comment]
Here's a new build integrating all of my PRs:
[ddev-intellij-plugin-0.0.1-dev.zip](https://github.com/user-attachments/files/16648849/ddev-intellij-plugin-0.0.1-dev.zip)

**Review by nico-loeber — APPROVED** [review]

File: build.gradle.kts [hunk 1]
```diff
@@ -39,10 +40,10 @@ dependencies {
 
     intellijPlatform {
         phpstorm(properties("platformVersion"))
-        pluginVerifier("1.307")
+        pluginVerifier("1.373")
         zipSigner()
         instrumentationTools()
-        testFramework(TestFrameworkType.Platform.JUnit4)
+        testFramework(TestFrameworkType.Platform)
```
Comment (nico-loeber) [inline build.gradle.kts:46]: Could you please elaborate on this change?
You added `testImplementation("junit:junit:4.13.2")` so we are still using junit4?

**Review by nico-loeber — CHANGES_REQUESTED** [review]
The plugin verifier task is currently failing due to some error.

```
Error: Exception in thread "main" com.jetbrains.plugin.structure.intellij.platform.InvalidIdeException: IDE path [/home/runner/.gradle/caches/transforms-4/5feadf3f0a3014be84dad66e5a46d47d/transformed/datagrip-2024.2] does not contain 'modules/module-descriptors.jar' file
	at com.jetbrains.plugin.structure.intellij.platform.BundledModulesResolver.<init>(BundledModulesResolver.kt:27)
	at com.jetbrains.plugin.structure.intellij.platform.BundledModulesResolver.<init>(BundledModulesResolver.kt:21)
	at com.jetbrains.plugin.structure.ide.ProductInfoBasedIdeManager.readPlugins(ProductInfoBasedIdeManager.kt:83)
	at com.jetbrains.plugin.structure.ide.ProductInfoBasedIdeManager.createIde(ProductInfoBasedIdeManager.kt:72)
	at com.jetbrains.plugin.structure.ide.ProductInfoBasedIdeManager.createIde(ProductInfoBasedIdeManager.kt:61)
	at com.jetbrains.plugin.structure.ide.DispatchingIdeManager.createIde(DispatchingIdeManager.kt:19)
	at com.jetbrains.plugin.structure.ide.DispatchingIdeManager.createIde(DispatchingIdeManager.kt:11)
	at com.jetbrains.pluginverifier.ide.IdeDescriptor$Companion.create(IdeDescriptor.kt:60)
	at com.jetbrains.pluginverifier.options.OptionsParser.createIdeDescriptor(OptionsParser.kt:114)
	at com.jetbrains.pluginverifier.options.OptionsParser.createIdeDescriptor(OptionsParser.kt:105)
	at com.jetbrains.pluginverifier.tasks.checkPlugin.DefaultIdeDescriptorParser.parseIdeDescriptors(CheckPluginParamsBuilder.kt:126)
	at com.jetbrains.pluginverifier.tasks.checkPlugin.CheckPluginParamsBuilder.build(CheckPluginParamsBuilder.kt:41)
	at com.jetbrains.pluginverifier.tasks.checkPlugin.CheckPluginParamsBuilder.build(CheckPluginParamsBuilder.kt:27)
	at com.jetbrains.pluginverifier.PluginVerifierMain$main$3$1.invoke(PluginVerifierMain.kt:130)
	at com.jetbrains.pluginverifier.PluginVerifierMain$main$3$1.invoke(PluginVerifierMain.kt:123)
	at com.jetbrains.pluginverifier.tasks.profiling.PluginVerificationProfilingsKt.measurePluginVerification(PluginVerificationProfilings.kt:11)
	at com.jetbrains.pluginverifier.PluginVerifierMain.main(PluginVerifierMain.kt:123)

```

Maybe this is not directly related to this pull request, but the plugin verifier is checking the compatibility to different IDE platforms and a positive result is mandatory for a release.

Also there is a deprecation notice that should be fixed:
```
/home/runner/work/ddev-intellij-plugin/ddev-intellij-plugin/src/main/java/de/php_perfect/intellij/ddev/database/DdevDataSourceManagerImpl.java:59: warning: [removal] performAutoSyncTask(Project,LocalDataSource) in DataSourceUtil has been deprecated and marked for removal
            DataSourceUtil.performAutoSyncTask(this.project, dataSource);
                          ^
Note: Some input files use or override a deprecated API.
```

**Comment by nico-loeber** [comment]
Great work so far @AkibaAT! I will take a look at the plugin verifier issue. Let me know if you already have an idea on how to solve this.

File: build.gradle.kts (hunk 1 above)
Comment (AkibaAT) [reply build.gradle.kts:46]: The rename is per https://github.com/JetBrains/intellij-platform-gradle-plugin/releases/tag/v2.0.0-beta7

The junit was added as per the breaking changes note on https://github.com/JetBrains/intellij-platform-gradle-plugin/releases/tag/v2.0.0-RC1

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -24,7 +24,7 @@ jobs:
         uses: actions/setup-java@v4
         with:
           distribution: zulu
-          java-version: 17
+          java-version: 21
```
[the remark this answers is not in the document]
Comment (AkibaAT) [reply .github/workflows/build.yml:27]: Since 2024.2 uses Java 21, I switched to 21 in all places I could find. That was the whole reason really.

**Comment by AkibaAT** [comment]
> Also there is a deprecation notice that should be fixed:
> 
> ```
> /home/runner/work/ddev-intellij-plugin/ddev-intellij-plugin/src/main/java/de/php_perfect/intellij/ddev/database/DdevDataSourceManagerImpl.java:59: warning: [removal] performAutoSyncTask(Project,LocalDataSource) in DataSourceUtil has been deprecated and marked for removal
>             DataSourceUtil.performAutoSyncTask(this.project, dataSource);
>                           ^
> Note: Some input files use or override a deprecated API.
> ```

I don't see any apparent replacement for this deprecation, and it seems to still be used the same within the database plugin.

**Comment by AkibaAT** [comment]
> The plugin verifier task is currently failing due to some error.
> 
> Maybe this is not directly related to this pull request, but the plugin verifier is checking the compatibility to different IDE platforms and a positive result is mandatory for a release.

I've reverted pluginVerifier to 1.307. The task completes successfully locally with that previous version. I could not get the current version to work properly, also because the -mute TemplateWordInPluginId (to ignore the jetbrains in the plugin ID) can't be set properly.

**Comment by AkibaAT** [comment]
3 deprecations are open:
- Deprecated method com.intellij.util.ui.UI.PanelFactory.panel(javax.swing.JComponent component) : com.intellij.openapi.ui.panel.ComponentPanelBuilder is invoked in de.php_perfect.intellij.ddev.settings.DdevSettingsComponent.<init>(Project)
  - Deprecated 2023-06-16: "ComponentPanelBuilder will be removed after moving Kotlin UI DSL into platform API package"
  - Unsure what the status is. Possibly resolvable with existing methods.
- Deprecated class com.intellij.openapi.ui.panel.ComponentPanelBuilder is referenced in de.php_perfect.intellij.ddev.settings.DdevSettingsComponent.<init>(Project)
  - Same as above
- Deprecated method com.intellij.database.util.DataSourceUtil.performAutoSyncTask(com.intellij.openapi.project.Project project, com.intellij.database.dataSource.LocalDataSource dataSource) : void is invoked in de.php_perfect.intellij.ddev.database.DdevDataSourceManagerImpl.lambda$updateDdevDataSource$2(LocalDataSourceManager, LocalDataSource) : void. This method will be removed in  a future release
  - Not part of the platform API, analysis of version before deprecation and current show no internal code migration to replicate as the deprecated function is still being used

**Comment by AkibaAT** [comment]
pluginVerifier 1.373 would generally work when adding
```
        freeArgs = listOf(
            "-mute",
            "TemplateWordInPluginId"
        )
```
_but_ it doesn't work with DataGrip, erroring out with `does not contain 'modules/module-descriptors.jar'`.

**Comment by nico-loeber** [comment]
The plugin verifier and other intellij platform gradle plugin tasks often cause trouble. Often a downgrade is the only fix.
Thanks for your great work @AkibaAT!

**Review by nico-loeber — APPROVED** [review]

**Pull request merged**